libbeat/processors/add_process_metadata/config.go

```diff
@@ -0,0 +1,126 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package add_process_metadata
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type config struct {
+	// IgnoreMissing: Ignore errors if event has no PID field.
+	IgnoreMissing bool `config:"ignore_missing"`
+
+	// OverwriteKeys allow target_fields to overwrite existing fields.
+	OverwriteKeys bool `config:"overwrite_keys"`
+
+	// RestrictedFields make restricted fields available (i.e. env).
+	RestrictedFields bool `config:"restricted_fields"`
+
+	// MatchPIDs fields containing the PID to lookup.
+	MatchPIDs []string `config:"match_pids"`
```

Comment:
Do you have an validation on this option? If not I'd add a tag for that `validate:"required"` and go-ucfg will do it.